### PR TITLE
terraform-provider-rhcs: Add 2 more test suits

### PR DESCRIPTION
**Pull request by amalykhi**
The following test suits are added:
1. Negative Periodic
2. Optional presubmit

### CODE REVIEW

**Comment by xueli181114** [comment]
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-optional-presubmit

**Comment by amalykhi** [comment]
/pj-rehearse  periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse  periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7
/pj-rehearse  pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-optional-presubmit
/pj-rehearse  pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-optional-presubmit
/pj-rehearse  periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse  periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-optional-presubmit
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-optional-presubmit
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-optional-presubmit
/pj-rehearse pull-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-optional-presubmit
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 1]
```diff
@@ -104,6 +104,66 @@ tests:
...
+  cron: 38 2 2,9,16,23,30 * *
+  steps:
+    cluster_profile: aws-terraform-qe
+    env:
+      CASE_LABEL_FILTER: day1-negative&&!Exclude
+      CHANNEL_GROUP: candidate
+      CLUSTER_PROFILE: rosa-sts-pl
+      REGION: us-east-1
+      RHCS_ENV: staging
+    post:
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:160]: For day 1 negative cases why not just remove pre and post?

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 2]
```diff
@@ -104,6 +104,66 @@ tests:
...
+    - ref: rhcs-general-teardown
+    pre:
+    - ref: rhcs-general-setup
+    test:
+    - ref: rhcs-general-tests
+    workflow: rhcs-aws-sts
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:166]: And also remove workflow, instead just has test

File: ci-operator/step-registry/rhcs/general/setup/rhcs-general-setup-commands.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+make tools
+make install
+
+# Below step will skip gcc checking
+export CGO_ENABLED=0
+
+# tar the shared manifest dir to make it share between pods
+cd ${SHARED_DIR}
+find ./tf-manifests -name 'terraform.[tfstate|tfvars]*' -print0|tar --null -T - -zcvf statefiles.tar.gz
```
Comment (xueli181114) [inline ci-operator/step-registry/rhcs/general/setup/rhcs-general-setup-commands.sh:68]: If there is no cluster creation, it's not meaningfull to tar the tfstate and tfvars file

File: ci-operator/step-registry/rhcs/general/teardown/rhcs-general-teardown-commands.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+export VERSION=${VERSION}
+export REGION=${REGION}
+
+make tools
+make install
+
+# Below step will skip gcc checking
+export CGO_ENABLED=0
+
+ginkgo run \
```
Comment (xueli181114) [inline ci-operator/step-registry/rhcs/general/teardown/rhcs-general-teardown-commands.sh:75]: we do not need a CLUSTER_PROFILE for negative testing right?

Comment (amalykhi) [reply ci-operator/step-registry/rhcs/general/teardown/rhcs-general-teardown-commands.sh:75]: Actually, for example for the cluster-admin cases, I need to prepare the  all the required resources (account/operator roles, oidc) to apply the cluster resource and get the expected error. SO I need different parameters from the profile. WDYT?

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml (hunk 2 above)
Comment (amalykhi) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml]: I'll try to run just tests

Comment (amalykhi) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml]: @xueli181114 Can't run just tests part, failing on states archive:
```
+ trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+ ls /tmp/secret
+ cp -r /root/terraform-provider-rhcs /alabama/
+ cp -r /alabama/terraform-provider-rhcs/tests/tf-manifests /tmp/secret/tf-manifests
+ cd /tmp/secret
+ tar -xvf statefiles.tar.gz
tar: statefiles.tar.gz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
```

Comment (amalykhi) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml]: @xueli181114 Do you think return to the general tests script but without the pre?
In post steps I would probably want to delete possible leftovers (account/operator roles, VPC, etc)

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-medium-low-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-medium-low-f7

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 5]
```diff
@@ -104,6 +104,56 @@ tests:
...
+  steps:
+    cluster_profile: aws-terraform-qe
+    env:
+      CASE_LABEL_FILTER: day1-negative&&!Exclude
+      CHANNEL_GROUP: candidate
+      CLUSTER_PROFILE: rosa-sts-ad
+      REGION: ap-northeast-1
+      RHCS_ENV: staging
+    test:
+    - ref: rhcs-e2e-general-tests
```
Comment (shellymiron) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:144]: question - I see that here (and in other places) there is no `workflow: rhcs-aws-sts` as it is done for other runs.
is that on purpose or should be placed here?

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml (hunk 2 above)
Comment (shellymiron) [reply ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml]: @xueli181114 why we don't need workflow? it holds the teardown no? without it the clusters does not
get deleted and stayed in the the staging env.

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 6]
```diff
@@ -104,6 +104,56 @@ tests:
...
+    env:
+      CASE_LABEL_FILTER: (Medium,Low)&&(day1-post,day2)&&!Exclude
+      CHANNEL_GROUP: candidate
+      CLUSTER_PROFILE: rosa-sts-pl
+      REGION: us-east-1
+      RHCS_ENV: staging
+    test:
+    - ref: rhcs-e2e-tests
+    workflow: rhcs-aws-sts
+- as: rosa-sts-advanced-negative-f7
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:133]: Let's make the title include day1-negative so we know what the job will run in the test.

File: ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml [hunk 7]
```diff
@@ -104,6 +104,56 @@ tests:
...
+    cluster_profile: aws-terraform-qe
+    env:
+      CASE_LABEL_FILTER: day1-negative&&!Exclude
+      CHANNEL_GROUP: candidate
+      CLUSTER_PROFILE: rosa-sts-ad
+      REGION: ap-northeast-1
+      RHCS_ENV: staging
+    test:
+    - ref: rhcs-e2e-general-tests
+- as: rosa-sts-private-negative-f7
```
Comment (xueli181114) [inline ci-operator/config/terraform-redhat/terraform-provider-rhcs/terraform-redhat-terraform-provider-rhcs-main.yaml:145]: So do the title, to make it clear about our testing scope

**Comment by nirarg** [comment]
Lets avoid changing the OWNERS file in this PR, to have it with less noise
In addition, lets try to have convention here (both commit and PR):

Name:
`terraform-provider-rhcs: Add 2 more test suits`

Description:
```
The following test suits are added:
1. Negative Periodic
2. Optional presubmit
```

**Comment by nirarg** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-negative-f7

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-private-day1-negative-f7
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-advanced-day1-negative-f7

**Pull request merged**